exporter/containerimage/writer.go

```diff
@@ -369,8 +389,14 @@ func (ic *ImageWriter) exportLayers(ctx context.Context, refCfg cacheconfig.RefC
 // the new blob.
 //
 // If no conversion is needed, this returns nil without error.
-func rewriteImageLayerWithEpoch(ctx context.Context, cs content.Store, desc ocispecs.Descriptor, comp compression.Config, epoch *time.Time) (*ocispecs.Descriptor, error) {
-	converterFn, err := converter.NewWithRewriteTimestamp(ctx, cs, desc, comp, epoch)
+func rewriteImageLayerWithEpoch(ctx context.Context, cs content.Store, desc ocispecs.Descriptor, comp compression.Config, epoch *time.Time, immDiffID digest.Digest) (*ocispecs.Descriptor, error) {
+	var immDiffIDs map[digest.Digest]struct{}
```

Comment:
I don't understand this construction in here. Why is a map with a single item passed? Instead of for example just setting epoch nil?

Reply:
> Why is a map with a single item passed? 

The `converter` pkg is designed to cover wider usecases that are not needed by `exporter/containerimage`.

> Instead of for example just setting epoch nil?

Because the converter may not make the final decision until computing the diff ID (not always available in the label)